### PR TITLE
Fix problems detected by pylint 2.8.1.

### PR DESCRIPTION
**Pull request by aarlt**

### CODE REVIEW

File: scripts/endToEndExtraction/remove-testcases.py [hunk 1]
```diff
@@ -118,40 +118,40 @@ def get_tests(e2e_path):
...
+        for line in cpp_file.readlines():
+            test = re.search(r'BOOST_AUTO_TEST_CASE\((.*)\)', line, re.M | re.I)
+            if test:
+                test_name = test.group(1)
+                inside_test = True
+                inside_extracted_test = inside_test & (test_name in tests)
+                if inside_extracted_test:
+                    count = count + 1
+
             if interactive and inside_extracted_test:
```
Comment (aarlt) [inline scripts/endToEndExtraction/remove-testcases.py:137]: Looks strange, but seem to be correct.

File: scripts/regressions.py [hunk 2]
```diff
@@ -67,13 +67,13 @@ def run_cmd(command, logfile=None, env=None):
...
-        proc = subprocess.Popen(command, shell=True, executable='/bin/bash',
-                                env=env, stdout=logfh,
-                                stderr=subprocess.STDOUT)
-        ret = proc.wait()
-        logfh.close()
-        return ret
+        with open(logfile, 'w') as logfh:
+            with subprocess.Popen(command, shell=True, executable='/bin/bash',
+                                    env=env, stdout=logfh,
+                                    stderr=subprocess.STDOUT) as proc:
```
Comment (hrkrshnn) [inline scripts/regressions.py:73]: Deviates from the indentation style in the file. Not too important.

**Review by hrkrshnn — APPROVED** [review]

File: scripts/endToEndExtraction/remove-testcases.py [hunk 3]
```diff
@@ -61,51 +61,51 @@ def get_checks(content, sol_file_path):
...
+        sol_checks = []
+        inside_expectations = False
+        for line in sol_file.readlines():
+            if line.startswith("// constructor()"):
+                sol_constructors.append(line)
+            elif inside_expectations and line.startswith("// "):
+                sol_checks.append(line)
+            if line.startswith("// ----"):
+                inside_expectations = True
+        sol_file.close()
```
Comment (cameel) [inline scripts/endToEndExtraction/remove-testcases.py:75]: This is no longer needed if you're using `with`. Also, the `with` block should really end here.

File: scripts/endToEndExtraction/remove-testcases.py [hunk 4]
```diff
@@ -61,51 +61,51 @@ def get_checks(content, sol_file_path):
...
-    what = ""
-    print("\nContinue? (ENTER) Abort? (ANY OTHER KEY)")
-    while what != '\n':
-        what = getkey()
-        if what != '\n':
-            sys.exit(0)
-    print()
+    with tempfile.NamedTemporaryFile(delete=False) as cpp_file:
+        cpp_file.write(content.encode())
+        cpp_file.close()
```
Comment (cameel) [inline scripts/endToEndExtraction/remove-testcases.py:85]: Again, `close` not needed with `with`.

**Pull request merged**